docs/testplan/transceiver/diagrams/data_flow.md

```diff
@@ -0,0 +1,199 @@
+    A --> E
+    B --> E
+    C --> E
+    D --> E
+    X --> T[Prerequisite Test Runner]
+    T --> O
+    
+    E --> F
+    F --> G
+    G --> H
+    H --> I
+    
+    I --> K
+    I --> L
+    I --> M
+    I --> N
+    
+    K --> J
+    L --> J
+    M --> J
+    N --> J
+    
+    J --> R
+    Q --> R
+    R --> P
+    P --> O
+    J --> O
+    
+    style A fill:#e1f5fe
+    style E fill:#f3e5f5
+    style J fill:#e8f5e8
+    style O fill:#fff3e0
+```
+
+## Detailed Processing Flow
+
+```mermaid
+sequenceDiagram
+    participant DI as dut_info.json
+    participant AM as AttributeManager
+    participant PP as Port Processor
+    participant CP as Config Parser
+    participant CF as Category Files
+    participant PR as Priority Resolver
+    participant PD as port_attributes_dict
+    participant V as Validator
+    participant TC as Test Cases
+    participant PRQ as Prerequisites File
+
+    TC->>AM: Initialize framework & load base data
+    AM->>DI: Load dut_info.json
+    DI-->>AM: Port specs & metadata
+    AM->>PP: Expand port specifications
+    PP-->>AM: Expanded port list
+    AM->>CP: Parse transceiver configuration strings
+    CP-->>AM: Parsed components
+    AM->>PD: Seed BASE_ATTRIBUTES per port
+
+    loop For each category
+        TC->>CF: Load category JSON file
+        CF-->>TC: Raw category attributes
+        TC->>PR: Resolve via 8-level priority hierarchy
+        PR-->>TC: Merged CATEGORY_ATTRIBUTES
+        TC->>PD: Store CATEGORY_ATTRIBUTES
+        opt Attribute Completeness Validation
+            PD->>V: Validate against deployment templates
+            V-->>PD: Results
+        end
+        TC->>PRQ: Load category prerequisite list (if present)
+        PRQ->>PRQ: Execute prerequisite tests (gating checks)
+        PRQ-->>TC: Continue if all pass
+        PD-->>TC: Run main category test cases
+    end
+```
+
+## Data Transformation Examples
+
+### Step 1: Port Expansion
+
+```text
+Input (dut_info.json):
+{
+  "dut_name_1": {
+    "Ethernet4:7": {
+      "vendor_name": "ACME Corp.",
+      "transceiver_configuration": "AOC-100-QSFPDD-2x100G_100G_SIDE-0xFF-0xFF"
+    }
+  }
+}
+
+After Port Expansion:
+- Ethernet4: same attributes
+- Ethernet5: same attributes  
+- Ethernet6: same attributes
+```
+
+### Step 2: Configuration Parsing
+
+```text
+Input: "AOC-100-QSFPDD-2x100G_100G_SIDE-0xFF-0xFF"
+
+Parsed Components:
+- cable_type: "AOC"
+- speed_gbps: 100
+- form_factor: "QSFPDD"
+- deployment: "2x100G_100G_SIDE"
+- media_lane_mask: "0xFF"
+- host_lane_mask: "0xFF"
+- media_lane_count: 8
+- host_lane_count: 8
+```
+
+### Step 3: Attribute Merging
+
+```text
+For Ethernet4 EEPROM_ATTRIBUTES:
+
+Priority Resolution:
+1. defaults.dual_bank_supported = false
+2. deployment_configurations.2x100G_100G_SIDE.dual_bank_supported = true  ← WINS
+3. vendor.ACME_CORP.defaults.dual_bank_supported = false
+4. No higher priority overrides found
+
+Result: dual_bank_supported = true
+```
+
+### Step 4: Final Structure
```

Comment:
@mihirpat1 did we add the debug log which can be turned ON when necessary to see this final structure?